include/dali/core/small_vector.h

```diff
@@ -248,6 +248,9 @@ class SmallVector : SmallVectorAlloc<T, allocator>, SmallVectorBase<T> {
   DALI_NO_EXEC_CHECK
   DALI_HOST_DEV void copy_assign(const T *data, size_t count) {
     clear();
+    if (!data) {
+      return;
+    }
```

Comment:
This is still broken. If the caller passes a null pointer and nonzero count, then the function is expected to reliably crash (however that sounds) rather than return quitely dismiss the error. This causes unexpected internal state:
- the data buffer is not changed
- the size is 0 instead of expected `count`
The subsequent code might attempt to access up to `count` elements which have unexpected values. This is far worse than a null pointer dereference which would simply kill the program.

Reply:
If you insist on having that check (which is superfluous), a viable alternative would be:
```suggestion
    if (!count)
      return;
#ifndef __CUDA_ARCH
      assert(data != nullptr);
#endif
```
(and remember to add `#include <cassert>` before `<utility>`)

Reply:
In face, I think this is a bad interface by design, we should accept a reference  and a size_t. We shouldn't care a null pointer in such a low level method. If the constructor or any other caller  called it, than it would have to throw an exception or simply crash.

Reply:
Let's not go this far in this PR. Also, it mimics what you can do in `std::string` and discussing whether or not std::string is a good design is a bit out of scope here.

Reply:
Ok, let's revert it.